### ISSUE TITLE
recovery after disk crash

### ISSUE THREAD

**Issue by yee379** [influxdata/influxdb#7042]
Influx 0.12.2
RHEL 6
XFS data partition

After a rather serious series of disks dying on my raid6 strip (3 out of 10), our storage admin were able to recover the partition that resided our influx data. xfs_check reports no issues.

However, when I start influxdb, I get the following error:

[filestore] 2016/07/21 00:52:56 /var/lib/influxdb/data/collectd/default/361/000000432-000000007.tsm (#1) opened in 6.776687ms
panic: runtime error: slice bounds out of range

I also tried deleting the wal directory under /var/lib/influxdb, but it results in the same error.

I assume that this was caused by some loss in data as a result of the raid failure.

My question is whether there is a way I can fsck the tsm files and perhaps purge all the bad ones so that i don't loose all of the time series data that I already have.

**Comment by mark-rushakoff** [comment]
The latest nightly should be able to handle corrupted TSM files without crashing, but it won't be able to read past the first corrupted part of the file.

**Comment by yee379** [comment]
Thanks @mark-rushakoff, does that mean that the corrupt tsm file will remain corrupt? Will I be able to read and write data once influx rolls over to a new tsm file?

**Comment by jwilder** [comment]
@yee379 The server should start up and not panic and you should see an error in the log about the problematic file that you can then move out of the directory.

For example,
```
[store] 2016/07/21 12:48:49 Failed to open shard: 3: [shard 3] error opening memory map for file /var/lib/influxdb/data/stress/autogen/3/000000001-000000001.tsm: mmapAccessor: invalid indexStart
```

**Comment by jwilder** [comment]
Fixed via #6995
